marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/CallBatcherTest.java

```diff
@@ -0,0 +1,336 @@
+		  batcher.onCallSuccess(event-> { 
+			  assignedParams.add(event.getArgs().getAssignedParamNames()[0]);
+		  });
+		  
+		  batcher.getDataMovementManager().startJob(batcher);
+		  
+	      batcher.add(oneCaller.args().param("param1", 1.2));
+
+	      batcher.flushAndWait();
+	      assertTrue(assignedParams.size()!=0);
+	      assertTrue(assignedParams.get(0).contains("param1"));
+	  }
+	  
+	  @Test
+	  public void manyForArgsTest() {
+		  CallManager.CallableEndpoint callableEndpoint = installEndpoint("manyParam");
+		  CallManager.ManyCaller<Float> manyCaller = callableEndpoint.returningMany(Float.class);
+		  
+		  CallBatcher<CallManager.CallArgs,CallManager.ManyCallEvent<Float>> batcher = manyCaller.batcher().forArgs();
+		  assertNotNull(batcher);
+		  Float[] values = new Float[]{1.8f, 2.4f};
+		  ArrayList<String> assignedParams = new ArrayList<String>();
+		  batcher.onCallSuccess(event-> { 
+			  assignedParams.add(event.getArgs().getAssignedParamNames()[0]);
+		  });
+		  
+		  batcher.getDataMovementManager().startJob(batcher);
+		  
+	      batcher.add(manyCaller.args().param("param1", values));
+
+	      batcher.flushAndWait();
+	      
+	      assertTrue(assignedParams.size()!=0);
+	      assertTrue(assignedParams.get(0).contains("param1"));
+	  }
+	  
+	    @AfterClass
+	    public static void release() {
+	        DatabaseClient adminClient = Common.newServerAdminClient("java-unittest-modules");
+	
+	        QueryManager queryMgr = adminClient.newQueryManager();
+	        DeleteQueryDefinition deletedef = queryMgr.newDeleteDefinition();
+	        deletedef.setDirectory(ENDPOINT_DIRECTORY);
+	        queryMgr.delete(deletedef);
+	
+	        adminClient.release();
+	       
+	    }
+	    
+	    private static void setupNoParamReturnEndpoint(
+	            JSONDocumentManager docMgr, DocumentMetadataHandle docMeta, String functionName, String datatype, String returnVal
+	    ) {
+	        JsonNode endpointdef = getEndpointdef(functionName, datatype, null, null, false, false);
+	        String script = getScript(datatype, null, null, false, false);
+	        setupEndpoint(docMgr, docMeta, endpointdef, script);
+	    }
+	    
+	    private static void setupEndpointMultipleRequired(JSONDocumentManager docMgr, DocumentMetadataHandle docMeta, String datatype, String functionName) {
+	        JsonNode endpointdef = getEndpointdef(functionName, datatype, true, false);
+	        String script = getScript(datatype, null, null, true, false);
+	        setupEndpoint(docMgr, docMeta, endpointdef, script);
+	    }
+	    
+	    private static void setupEndpointSingleRequired(
+	            JSONDocumentManager docMgr, DocumentMetadataHandle docMeta, String functionName, String datatype
+	    ) {
+	        JsonNode endpointdef = getEndpointdef(functionName, datatype, false, false);
+	        String script = getScript(datatype,null,null, false, false);
+	        setupEndpoint(docMgr, docMeta, endpointdef, script);
+	    }
+	    
```

Comment:
For the sake of more maintainability, could the endpoint creation methods in CallManagerTest become utilities that could be called from either CallManagerTest or CallBatcherTest?  

Likely those functions use some fields of CallManagerTest.  Maybe those fields could move into a static nested class of CallManagerTest, the shared functions could become methods of that static nested class, and CallManagerTest and CallBatcherTest could each have an instance of that static nested class that's used for setup.

What do you think?

Reply:
Yes that's right. 
Moving the methods to the static nested class EndpointSetup.

Reply:
The nested class has dependencies on CallManagerTest.

If both CallManagerTest and CallBatcherTest construct an instance of the setup class, could each configure the instance for its needs?  For instance if the constructor is passed in the endpoint directory, could the constructor of the nested setup class create the service definition JsonNode as a field such that the methods (if the methods of the setup class become object methods instead of static methods) can have access to the field?